src/de/mod.rs

```diff
@@ -126,21 +133,27 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         V: Visitor<'de>,
     {
         if self.bytes.consume_ident("true") {
-            return visitor.visit_bool(true);
+            return visitor.visit_bool(true).map_err(|e| self.fix_position(e));
```

Comment:
why all these cases just don't return positional errors to begin with?

Reply:
If I understand correctly, the errors are generated by `fn custom<T: fmt::Display>(msg: T) -> Self` on `Error` in `error.rs`.  In serde's json deserializer ( https://github.com/serde-rs/json/blob/master/src/error.rs#L380 ) they've already got the line number and column built into the message, somehow, but I have a hard time following the path.